source/MaterialXView/Viewer.cpp

```diff
@@ -137,6 +137,12 @@ void applyModifiers(mx::DocumentPtr doc, const DocumentModifiers& modifiers)
     }
 }
 
+bool useFloatBuffer()
+{
+    auto [capability_10bit, capability_EDR] = ng::test_10bit_edr_support();
```

Comment:
ng treats a request for a float buffer as a request for both a float buffer, and an EDR layer, however if the hardware can't do it, the float buffer will still work fine as the EDR status is just a request. Over unity values will get clamped in that case, but this viewer doesn't do HDR rendering, so you won't see the effect of that.

This ng function call tells you if *any* connected NSScreen is EDR or 10 bit capable, not necessarily the monitor your window is on...  Since the window manager can deal with 16 bit and edr frame buffers irrespective of the device capabilities, I would personally suggest just running a half frame buffer all the time.

Reply:
Good info! It would certainly simplify the code to always use 16 bit buffers, so if everyone is fine with that then I can make that change.

Reply:
Having said that, there would probably need to be changes elsewhere to support 16 bit across the board. Since this PR is focused on macOS, we could limit float buffers to that platform (perhaps via `MATERIALXVIEW_METAL_BACKEND`) for now.